pkg/util/merger.go

```diff
@@ -19,12 +19,8 @@ func MergeSampleSets(a, b []model.SamplePair) []model.SamplePair {
 			j++
 		}
 	}
-	for ; i < len(a); i++ {
-		result = append(result, a[i])
-	}
-	for ; j < len(b); j++ {
-		result = append(result, b[j])
-	}
+	result = append(result, a[i:]...)
```

Comment:
I think this deserves a comment like "now add the tail of either a or b; at least one of these appends is a no-op as a result of the loop condition"